s3fs/core.py

```diff
@@ -12,14 +12,21 @@
 
 from fsspec.spec import AbstractBufferedFile
 from fsspec.utils import infer_storage_options, tokenize, setup_logging as setup_logger
-from fsspec.asyn import AsyncFileSystem, sync, sync_wrapper, FSTimeoutError
+from fsspec.asyn import (
+    AsyncFileSystem,
+    sync,
+    sync_wrapper,
+    FSTimeoutError,
+    _run_coros_in_chunks,
```

Comment:
I don't love this private import, so I'm given to just vendor this function over here. What do you think @martindurant?

Reply:
We could even expose it as a public function directly from fsspec.